### PR TITLE
Task 7

### PR DESCRIPTION
**Pull request by lai1372**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -208,3 +208,38 @@ describe("GET - /api/reviews/:review_id/comments", () => {
       });
   });
 });
+describe("POST - /api/reviews/:review_id/comments", () => {
+  test("should return a 201 along with the posted comment", () => {
+    return request(app)
+      .post("/api/reviews/3/comments")
+      .send({ username: "mallionaire", body: "lorem ipsum!" })
+      .expect(201)
+      .then((comment) => {
```
Comment (JVal-NC) [inline __tests__/app.test.js:217]: Can we fix this up so we dont have to do comment.body.comment.

Also Lines 218 and 220 are not needed if we're asserting against the property data types on lines 219 and 221 respectively.

File: __tests__/app.test.js [hunk 2]
```diff
@@ -208,3 +208,38 @@ describe("GET - /api/reviews/:review_id/comments", () => {
...
+      .send({ username: "mallionaire", body: "lorem ipsum!" })
+      .expect(404)
+      .then((response) => {
+        expect(response.body.msg).toBe(
+          "404 - no such review ID found, please use an existing review ID to post a comment"
+        );
+      });
+  });
+ 
+  test("should return a 400 if the wrong values are sent in the post request", () => {
```
Comment (JVal-NC) [inline __tests__/app.test.js:236]: Good tests, I would recommend a test for a username string that doesnt exist in the users table.

File: models/comments.models.js [hunk 3]
```diff
@@ -15,3 +15,33 @@ exports.fetchCommentsByReviewID = (reviewId) => {
...
+  .then((review) => {
+    if (review.rows.length === 0) {
+      return Promise.reject({
+        status: 404,
+        msg: "404 - no such review ID found, please use an existing review ID to post a comment"
+      });
+    }
+    const author = comment.username
+    const body = comment.body
+    if (typeof author !== "string"){
```
Comment (JVal-NC) [inline models/comments.models.js:33]: I believe psql will give us an error for this, lets use that error to handle this.

**Review by JVal-NC — COMMENTED** [review]
Really good work so far, some extra tests and a refactor of the model is all that is required.

**Review by elijaharley — APPROVED** [review]

**Pull request merged**